tests/shared/s3-packages.spec.js

```diff
@@ -0,0 +1,988 @@
+              <ETag>"1bafcc0d5d2c8d43bd2ce8948bb51a8b"</ETag>
+              <Size>57331229</Size>
+              <StorageClass>STANDARD</StorageClass>
+            </Contents>
+            <Contents>
+              <Key>notEtcher/1.0.0-beta.18/Etcher-1.0.0-beta.18-win32-x64.exe</Key>
+              <LastModified>2017-01-17T05:48:02.000Z</LastModified>
+              <ETag>"b0a6154ec79d17618632ac62eb30b44e"</ETag>
+              <Size>84802632</Size>
+              <StorageClass>STANDARD</StorageClass>
+            </Contents>
+            <Contents>
+              <Key>notEtcher/1.0.0-beta.18/Etcher-1.0.0-beta.18-win32-x64.zip</Key>
+              <LastModified>2017-01-17T06:14:46.000Z</LastModified>
+              <ETag>"f25c5dfa8378e608c25fafbe65e90162"</ETag>
+              <Size>82235087</Size>
+              <StorageClass>STANDARD</StorageClass>
+            </Contents>
+            <Contents>
+              <Key>notEtcher/1.0.0-beta.18/Etcher-1.0.0-beta.18-win32-x86.exe</Key>
+              <LastModified>2017-01-17T06:42:58.000Z</LastModified>
+              <ETag>"183b6eb648b0a78a1e99c9182f90194e"</ETag>
+              <Size>74264232</Size>
+              <StorageClass>STANDARD</StorageClass>
+            </Contents>
+            <Contents>
+              <Key>notEtcher/1.0.0-beta.18/Etcher-1.0.0-beta.18-win32-x86.zip</Key>
+              <LastModified>2017-01-17T07:05:55.000Z</LastModified>
+              <ETag>"d8d860013f038bed3f52534053b08382"</ETag>
+              <Size>72042525</Size>
+              <StorageClass>STANDARD</StorageClass>
+            </Contents>
+          </ListBucketResult>
+        `);
+      });
+
+      afterEach(function() {
+        nock.cleanAll();
+      });
+
+      it('should resolve an empty array', function(done) {
+        s3Packages.getRemoteVersions(s3Packages.BUCKET_URL.PRODUCTION).then((versions) => {
+          m.chai.expect(versions).to.deep.equal([]);
+          done();
+        }).catch(done);
+      });
+
+    });
+
+    describe('given an unsuccessful request', function() {
+
+      beforeEach(function() {
+        nock(s3Packages.BUCKET_URL.PRODUCTION).get('/').reply(500);
+      });
+
+      afterEach(function() {
+        nock.cleanAll();
+      });
+
+      it('should be rejected with an error', function(done) {
+        s3Packages.getRemoteVersions(s3Packages.BUCKET_URL.PRODUCTION).catch((error) => {
+          m.chai.expect(error).to.be.an.instanceof(Error);
+          done();
+        });
+      });
+
+    });
+
+    describe('given ENOTFOUND', function() {
+
+      beforeEach(function() {
+        const error = new Error('ENOTFOUND');
+        error.code = 'ENOTFOUND';
+
+        this.requestGetAsyncStub = m.sinon.stub(request, 'getAsync');
+        this.requestGetAsyncStub.returns(Bluebird.reject(error));
+      });
+
+      afterEach(function() {
+        this.requestGetAsyncStub.restore();
+      });
+
+      it('should return an empty array', function(done) {
```

Comment:
Is there any reason for line 572 saying `'should resolve an empty array'` and this line saying `'should return an empty array'` ?